src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -253,58 +274,109 @@ pub struct RSIndexResult {
     pub weight: f64,
 }
 
+impl Default for RSIndexResult {
+    fn default() -> Self {
+        Self::virt(0, 0, 0.0)
+    }
+}
+
 impl RSIndexResult {
-    /// Create a new numeric index result with the given numeric value
-    pub fn numeric(doc_id: t_docId, num: f64) -> Self {
+    /// Create a new virtual index result
+    pub fn virt(doc_id: t_docId, field_mask: FieldMask, weight: f64) -> Self {
```

Comment:
Shouldn't we remove `pub fn freqs_only()` then? It's just another virt constructor.